compiler-core/src/build/module_loader.rs

```diff
@@ -20,6 +22,35 @@ use crate::{
     Error, Result,
 };
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
+pub(crate) struct SourceDigest(Output<Sha256>);
```

Comment:
Can we use the word "fingerprint" rather than "digest"? We may use digest for other things :)